docs/proposal/2022-06-14-configuration-bindings.md

```diff
@@ -0,0 +1,647 @@
+
+    Related issues:
+      - [Define the channel at the namespace level](https://github.com/infracloudio/botkube/issues/486)
+
+3. Defining `kubectl` (executor) permissions per channel—in particular, configuring what commands can be executed and in which Namespaces.
+
+    Currently, it's possible only if you will deploy multiple BotKube instances with different configurations.
+
+    Related issues:
+      - [Multiple Slack Channels](https://github.com/infracloudio/botkube/issues/250)
+
+### Non-goal
+
+Those are not the goals of this proposal. However, we should be able to implement them later.
+
+- [Notify groups](https://github.com/infracloudio/botkube/issues/323)
+- [Group error messages to thread](https://github.com/infracloudio/botkube/issues/545)
+- [Customizable Messages](https://github.com/infracloudio/botkube/issues/434)
+- [Support for default cluster and namespace on any Slack channels](https://github.com/infracloudio/botkube/issues/421)
+
+## Proposal
+
+### Terminology
+
+| Name           | Description                                                                                                                                                            |
+|----------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Executors      | Executes BotKube or `kubectl` command and sends back the result to the Bot. In the future, new executors might be added. For example, `helm`, `argo`, `istioctl`, etc. |
+| Sources        | Provides domain specific notification. For example, Kubernetes events, security events (Sysdig), metrics (Prometheus), and similar.                                    |
+| Communications | Describes both Bot and Sink.                                                                                                                                           |
+| Bot            | Bi-directional communication such as Slack, Discord, Mattermost.                                                                                                       |
+| Sink           | Unidirectional communication, such as Elasticsearch or Webhook. It may receive events from Sources only.                                                               |
+
+![](../assets/ideation-view.png#gh-light-mode-only)
+![](../assets/ideation-view-dark.png#gh-dark-mode-only)
+
+### New syntax
+
+This section describes the necessary changes in the syntax. **It's not backward compatible.**
+
+1. Sources
+    <table>
+    <tr>
+    <td> Before </td> <td> After </td>
+    </tr>
+    <tr>
+    <td>
+
+    ```yaml
+
+
+
+
+
+
+
+
+
+    resources:
+      - name: v1/pods
+        namespaces:
+          include: ["istio-system"]
+        events: ["error"]
+      - name: networking.istio.io/v1alpha3/VirtualServices
+        namespaces:
+          include: ["@all"]
```

Comment:
why `@all` instead of simply `all`?

Reply:
Because when we use `all`, we block an option of using such name for Namespaces. With `@all` you can see that it is a special tag based on the "API". Do you think this is not a necessary change? We thought that as we already change the API it will be good to straight this out too.

Reply:
wow that could be a real issue!